packages/indexer-agent/src/utils/balance-monitor.ts

```diff
@@ -0,0 +1,39 @@
+import { Gauge } from 'prom-client'
+import { Wallet, utils } from 'ethers'
+
+import { Logger, Metrics, timer } from '@graphprotocol/common-ts'
+import { indexerError, IndexerErrorCode } from '@graphprotocol/indexer-common'
+
+interface EthBalanceMetrics {
+  operatorEthBalance: Gauge<string>
+}
```

Comment:
You can avoid writing this interface by using `ReturnType<typeof registerMetrics>` where needed (example in [studio agent](https://github.com/edgeandnode/subgraph-studio/blob/main/packages/agent/src/jobs/settle-invoices.ts#L215))

Reply:
In fact... I don't even need a type at all here, since TypeScript will infer it from the return type. Good point.